frontend/hooks/data-fetching/course.ts

```diff
@@ -96,50 +104,114 @@ export async function getSemesters(): Promise<Semester[]> {
+        `/api/courses/${courseId}/queues/${queueId}/questions`,
+        (id) => `/api/courses/${courseId}/queues/${queueId}/questions/${id}/`,
+        {
+            model: "ohq.Question",
+            property: "queue_id",
+            value: queueId,
+        },
+        {
+            initialData,
+            fetcher: newResourceFetcher,
+            orderBy: (q1, q2) => {
+                const date1 = new Date(q1.timeAsked);
+                const date2 = new Date(q2.timeAsked);
+
+                if (date1 > date2) {
+                    return 1;
+                } else {
+                    return -1;
+                }
+            },
+        }
     );
+    const filteredData = data?.filter(
+        (q) =>
+            q.status === QuestionStatus.ACTIVE ||
+            q.status === QuestionStatus.ASKED
+    );
+    return { data: filteredData, ...other };
+};
 
 export const useQuestionPosition = (
     courseId: number,
     queueId: number,
-    id: number,
-    refreshInterval: number
-) =>
-    useResource(
+    id: number
+) => {
+    const { data: qdata } = useRealtimeResource<Queue>(
+        `/api/courses/${courseId}/queues/${queueId}/id`,
+        {
+            model: "ohq.Queue",
+            property: "id",
+            value: queueId,
+        },
+        { fetcher: newResourceFetcher }
+    );
+
+    const [
+        data,
+        error,
+        isValidating,
+        mutate,
+    ] = useResource(
         `/courses/${courseId}/queues/${queueId}/questions/${id}/position/`,
-        { position: -1 },
-        { refreshInterval }
+        { position: -1 }
     );
 
-export const useLastQuestions = (
-    courseId: number,
-    queueId: number,
-    refreshInterval: number
-) =>
-    useResourceList<Question>(
+    const stringified = JSON.stringify(qdata);
+    useEffect(() => {
+        mutate();
+    }, [stringified]);
+    return [data, error, isValidating, mutate];
+};
+
+// only student queues should use this, since it doesn't make
+// much sense otherwise
+export const useLastQuestions = (courseId: number, queueId: number) => {
+    const { data: qdata } = useRealtimeResourceList<Question, "queue_id">(
+        `/api/courses/${courseId}/queues/${queueId}/questions`,
+        (id) => `/api/courses/${courseId}/queues/${queueId}/questions/${id}`,
+        {
+            model: "ohq.Question",
+            property: "queue_id",
+            value: queueId,
+        },
+        { fetcher: newResourceFetcher }
+    );
+
+    const [data, error, isValidating, mutate] = useResourceList(
         `/courses/${courseId}/queues/${queueId}/questions/last/`,
-        (id) => `/courses/${courseId}/queues/${queueId}/last/${id}/`,
-        [],
-        { refreshInterval }
+        (id) => `/courses/${courseId}/queues/${queueId}/last/${id}/`
     );
 
+    const stringified = JSON.stringify(qdata);
+
+    // this revalidates the last question query whenever there is a websocket update
+    useEffect(() => {
```

Comment:
mutate is stale in this effect, this is fine because we only use it for revalidation, but it is bad style